### ISSUE TITLE
Can't assign string to dates in v2.9.9.

### ISSUE THREAD

**Issue by whataboutpereira** [uvarov-frontend/vanilla-calendar-pro#297]
```
self.settings.selected.dates = [this.inputTarget.value];
```

now causes

```
TS2322: Type 'string' is not assignable to type 'number | Date | `${number}-10-01`
```

Probably because ISelected

```
dates?: Array<Date | number | string>;
```

changed to

```
dates?: Array<Date | number | FormatDateString>;
```

Back to v2.9.8 because of BC break. :)

**Comment by whataboutpereira** [comment]
Of course it can be forces with  as FormatDateString.

**Comment by ghiscoding** [comment]
yeah I also saw this, we need to cast it more often since it changed from `string` to `FormatDateString` and the intelliSense is a little hard to follow and read when it fails (as shown below). Nonetheless, it's probably the correct type to use anyway, though I always wondered if there was an easier Type that we could use? 

![image](https://github.com/user-attachments/assets/c53084c7-1179-497a-b4a3-ae117883ecb2)

**Comment by uvarov-frontend** [comment]
‘FormatDateString’ is the correct data type for dates, import it from the plugin and use it instead of ‘string’.

**Comment by uvarov-frontend** [comment]
https://vanilla-calendar.pro/docs/reference/additionally/settings#selected-dates
